engine/src/main/java/org/hibernate/search/spi/SearchIntegratorBuilder.java

```diff
@@ -324,6 +326,8 @@ private void initDocumentBuilders(SearchConfiguration searchConfiguration, Build
 		final org.hibernate.search.engine.metadata.impl.MetadataProvider metadataProvider =
 				new AnnotationMetadataProvider( searchConfiguration.getReflectionManager(), configContext );
 
+		bindFilterDefsInSuperClasses( classMappings );
```

Comment:
See here you are processing now the filter definitions, after creating creating the `AnnotationMetadataProvider`. It could do the job of discovering the filter definitions. However, you would need to defer the retrieval of the filter definitions until `metadataProvider.getTypeMetadataFor( mappedClass )`, since the parsing of annotations is on request. See line 348 and 385. One would need to add the filter defs to `TypeMetadata`, which is maybe a bit fishy. `TypeMetadata` is supposed to hold all information related to a given type and it hierarchy, but the filer definitions are as mentioned type independent. That said, they are defined within this type hierarchy, so I guess we could add it. They need to be aggregated within `SearchIntegratorBuilder` then anyways.